### PR TITLE
Update bandc_c+a.rb

### PR DESCRIPTION
**Pull request by irhen**
+def
still having problems with breaking out of if's
also gets.chomp.downcase isn't working >.<
need some help over here?..

### CODE REVIEW

File: bandc_c+a.rb [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+
+def advanced_check(variable)
+  if variable.size != 6
+    puts "Please, try again. Your guess must contain exactly 6 digits!"
+  end
+        
+  if variable.uniq.length != 6
+    puts "Please, try again. Your guess must contain 6 unique digits!"
+  end
+end
```
Comment (maoeye18) [inline bandc_c+a.rb:26]: Almost same function as classic_check. Try use parameter for game type, and may be hash with   details of game: size, flags with needed checks, etc.

File: bandc_c+a.rb [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+
+while game_type == "c" or "a" do
```
Comment (maoeye18) [inline bandc_c+a.rb]: Shouldn't be `game_type == "c" or game_type == "a"`? But, even so looks strange. What about asking game type inside of loop?

File: bandc_c+a.rb [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
   if game_type == "c"  
     secret_number = Array(1..9).sample(4)
+    p secret_number #delete after testing
```
Comment (maoeye18) [inline bandc_c+a.rb]: It's better to clean up code before pushing.

File: bandc_c+a.rb [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
+    puts "Please, try again. Your guess must contain exactly 4 digits!"
+  end
+        
+  if variable.uniq.length != 4
+    puts "Please, try again. Your guess must contain 4 unique digits!"
+  end
+    
+  if variable.include?(0)
+    puts "Please, try again. You cannot use zero in your guess!"
+  end
```
Comment (maoeye18) [inline bandc_c+a.rb:15]: Your function doesn't return any result. Is it ok?

File: bandc_c+a.rb [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
-      end
-        
-      if user_guess.uniq.length != 4
-        puts "Please, try again. Your guess must contain 4 unique digits!"
-      end
-    
-      if user_guess.include?(0)
-        puts "Please, try again. You cannot use zero in your guess!"
-      end
+      classic_check(user_guess)
```
Comment (maoeye18) [inline bandc_c+a.rb:39]: You don't use result of function. It just puts some lines and here is no any action with results of check. Is it ok?

File: bandc_c+a.rb [hunk 6]
```diff
@@ -34,33 +50,28 @@
   
   elsif game_type == "a"
```
Comment (maoeye18) [inline bandc_c+a.rb]: So, classic and advanced games have almost same logic. Try to rewrite them in one method and give parameter with type of game. You may keep features of game types in hash, for example. Is there something like constants in ruby?

File: bandc_c+a.rb (hunk 4 above)
Comment (irhen) [reply bandc_c+a.rb:15]: i... guess so? is it not? :D

File: bandc_c+a.rb (hunk 5 above)
Comment (irhen) [reply bandc_c+a.rb:39]: it's really just for checking if there is any mistake in user's input. if there is, it prints some messages. :D

File: bandc_c+a.rb (hunk 6 above)
Comment (irhen) [reply bandc_c+a.rb]: there is. i'll think about it >.>

File: bandc_c+a.rb (hunk 2 above)
Comment (irhen) [reply bandc_c+a.rb]: doesn't matter really. both ways are correct, mine is shorter though.
don't really understand how to ask that INSIDE the loop when it's the CONDITION of the loop :) any ideas? another loop?

File: bandc_c+a.rb (hunk 3 above)
Comment (irhen) [reply bandc_c+a.rb]: done, ty

**Comment by irhen** [comment]
what about my gets.chomp.downcase? what could possibly go wrong with that?

File: bandc_c+a.rb (hunk 2 above)
Comment (maoeye18) [reply bandc_c+a.rb]: Loop without condition? Infinity one :) But well, no, it's not an option, bad practice.
Think about this ways:
1) Ask about game type in method and use it result in loop condition.
2) Ask game type twice, before loop and inside, check variable. This way is worse, i guess, but may be usefull for your learning and training.

File: bandc_c+a.rb (hunk 4 above)
Comment (maoeye18) [reply bandc_c+a.rb:15]: Well, if you are deciding to skip or not to skip an iteration, then you need some boolean reslut, don't you?

File: bandc_c+a.rb (hunk 5 above)
Comment (maoeye18) [reply bandc_c+a.rb:39]: And what is about skipping of loop iteration?

**Comment by maoeye18** [comment]
I don't see any `downcase`, what exactly looks wrong?

**Comment by irhen** [comment]
if you add .downcase to any gets.chomp, it just doesn't work. like, game_type = gets.chomp.downcase doesnt really respond to C or A. why is that?

File: bandc_c+a.rb (hunk 4 above)
Comment (irhen) [reply bandc_c+a.rb:15]: yeah. >.>

**Pull request merged**